ui/perfherder/compare/CompareTableView.jsx

```diff
@@ -166,25 +154,13 @@ export default class CompareTableView extends React.Component {
     this.setState({ timeRange }, () => this.getPerformanceData());
   };
 
-  notify = (message, severity, options = {}) => {
-    const { notifications } = this.state;
-    const notification = {
-      ...options,
-      message,
-      severity: severity || 'info',
-      created: Date.now(),
-    };
-    const newNotifications = [notification, ...notifications];
-
-    this.setState({
-      notifications: newNotifications,
-    });
-  };
-
-  clearNotification = index => {
-    const { notifications } = this.state;
-
-    this.setState(clearNotificationAtIndex(notifications, index));
+  notifyFailure = (message, severity) => {
+    const { failureMessages } = this.state;
+    if (severity === 'danger') {
+      this.setState({
+        failureMessages: [message, ...failureMessages],
```

Comment:
This notify function is a unnecessary. You can set the error messages based on `failureStatus` from the action performed in `CompareTableControls`, and use the `updateState` callback prop to set state.

Reply:
I used this notify function because the retrigger function from JobModel does not pass any errors back. It just catches them and uses this notify function to handle them.
Cameron mentioned that this will be changed soon (https://bugzilla.mozilla.org/show_bug.cgi?id=1571643#c10), but until then, I don't think we can do it in another way.